DuckDuckGo/Freemium/PIR/FreemiumDBPPromotionViewCoordinator.swift

```diff
@@ -0,0 +1,174 @@
+//
+//  FreemiumDBPPromotionViewCoordinator.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Combine
+import Foundation
+import Freemium
+import OSLog
+
+/// Protocol defining the interface for coordinating the visibility and interaction with the
+/// Freemium DBP (Data Broker Protection) promotion view.
+@MainActor
+protocol FreemiumDBPPromotionViewCoordinator: ObservableObject {
```

Comment:
The pattern we usually follow for protocol naming is to use the gerund form describing the purpose they serve, i.e. `FreemiumDBPPromotionViewCoordinating` (actually the same as `FreemiumPIRScanResultPolling` introduced in this very PR).

Reply:
Cool. Yeah, we should try to be consistent across the codebase. I’ve previously (including in this feature branch) used `ProtocolName` and then `DefaultProtocolName` instead of the gerund form. However I’ll update this to the gerund form and then stick to gerund form going forward.